ddtrace/opentracer/tracer.py

```diff
@@ -84,7 +99,7 @@ def __init__(self, service_name=None, config=None, scope_manager=None, dd_tracer
         dd_context_provider = get_context_provider_for_scope_manager(self._scope_manager)
 
         self._dd_tracer = dd_tracer or ddtrace.tracer or DatadogTracer()
-        self._dd_tracer.set_tags(self._config.get(keys.GLOBAL_TAGS))
+        self._dd_tracer.set_tags(self._config.get(keys.GLOBAL_TAGS))  # type: ignore[arg-type]
```

Comment:
we could also cast here

Reply:
Yeah this is also pretty messy since the problem is trying to grab the global tags dictionary from the 
 `DEFAULT_CONFIG` dictionary which is of type `Dict[str, Any]` as it contains strings, bools, ints, and another dict (the Global tags dict within `DEFAULT_CONFIG`)... one solution is to just add this assertion:
```suggestion
        global_tags = self._config.get(keys.GLOBAL_TAGS)
        assert isinstance(global_tags, dict)
        self._dd_tracer.set_tags(global_tags)
```
What do you think?

Reply:
if it is always going to be there + a dict... we can just do `typing.cast(Dict[str, Any], self._config.get(keys.GLOBAL_TAGS))`

Reply:
Just tried using `typing.cast(...)`, mypy doesn't like nested dictionaries with different types for some reason... the `typing.TYPE_CHECKING` assertion does the exact same thing and only at type-check time (not runtime) so we should be okay with this :)

Reply:
👍🏻